src/utils/request-utils.ts

```diff
@@ -42,8 +42,45 @@ export const formDataToQuery = function (formdata: Record<string, any> | FormDat
 }
 
 export const getQueryParam = function (url: string, param: string): string {
+    // Check if URL and URLSearchParams are available in the global scope
+    // Fall back to manual parsing if URL/URLSearchParams aren't supported
+    if (typeof URL === 'undefined' || typeof URLSearchParams === 'undefined') {
+        return getQueryParamFallback(url, param)
```

Comment:
Is there any downside to just always calling that function? It's nicer to use the built-in parsing, and if we could guarantee that it was available, we'd just use that, but given we have to have a fallback anyway, why add extra bytes to the bundle with two implementations?

Reply:
Just because I'd prefer to use the built-in if it exists, but I'm fine with keeping the old-but-proven-to-work function instead, and just fix it (which I did anyway)

Reply:
Which is basically what I commented here: https://github.com/PostHog/posthog-js/pull/1778#discussion_r1977941224

Reply:
Would you prefer we kept that old code instead and never add the new one?